File: python/04_008.py
class UserService:
  """
  Service for user operations.
  """
  def __init__(self, user_repository):
     self.user_repository = user_repository
  
  def get_user(self, username):
     return self.user_repository.find_by_username(username)
  
  def create_user(self, user):
     if self.user_repository.exists_by_username(
           user.get_username()):
        raise UserAlreadyExistsException()
     self.user_repository.save(user)

File: data structures and algorithms/cs/04_008.cs
class UserService
{
   private readonly UserRepository userRepository;

   public UserService(UserRepository userRepository)
   {
      this.userRepository = userRepository;
   }

   public User GetUser(string username)
   {
      return userRepository.FindByUsername(username);
   }

   public void CreateUser(User user)
   {
      if (userRepository.ExistsByUsername(user.Username))
      {
         throw new UserAlreadyExistsException();
      }
      userRepository.Save(user);
   }
}
